explorer/src/Pos/Explorer/Web/ExtraContext.hs

```diff
@@ -0,0 +1,48 @@
+-- Classes and datatypes for managing extra context in Explorer.
+
+module Pos.Explorer.Web.ExtraContext
+    ( ExtraContext
+    , ExtraContextT
+    , runExtraContextT
+    , makeExtraCtx
+
+    , HasGenesisRedeemAddressInfo (..)
+    , GenesisRedeemAddressInfo
+    ) where
+
+import           Universum
+
+import qualified Data.Vector   as V
+import qualified Ether
+
+import           Pos.Context   (NodeContext)
+import           Pos.Core      (Address, Coin, isRedeemAddress)
+import           Pos.Genesis   (GenesisUtxo (..))
+import           Pos.Txp       (utxoToAddressCoinPairs)
+import           Pos.Util.Util (lensOf)
+
+type ExtraContextT m = Ether.ReaderT' ExtraContext m
```

Comment:
Why a regular `ReaderT` isn't sufficient here? 🤔

Reply:
Because it is used in the context of

```haskell
type ExplorerProd = ExtraContextT (ExplorerBListener (RealMode SscGodTossing))
```

and

```haskell
type RealMode ssc = Mtl.ReaderT (RealModeContext ssc) Production
```

so there would be a conflict between two `ReaderT`s.